domain/secret/watcher_test.go

```diff
@@ -279,6 +279,57 @@ func (s *watcherSuite) TestWatchObsoleteForUnitsOwned(c *gc.C) {
 	wCSingleUnit.AssertNoChange()
 }
 
+func (s *watcherSuite) TestWatchObsoleteForUserSecrets(c *gc.C) {
```

Comment:
Are we missing a test where auto prune was false, and then got set to true with some obsolete revisions already in existence

Reply:
The uri1 is exactly for this test case.